File: 3379-transformed-array/3379-transformed-array.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        visited = set()
        result = nums[:]
        
        for i in range(len(nums)):
            ind = 0
            ind = (i+nums[i])%len(nums)
    
            result[i] = nums[ind]

        return result

File: 3379-transformed-array/3379-transformed-array.cpp
class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {

    int n = nums.size();
    vector<int> result(n);

    for (int i=0; i < n; i++){
        int ind = ((nums[i]+i)%n+n)%n;
        result[i] = nums[ind];
    }
        
        return result;
    }
};